doc/source/whatsnew/v1.2.0.rst

```diff
@@ -677,6 +677,7 @@ Indexing
 - Bug in :meth:`DataFrame.loc` and :meth:`DataFrame.__getitem__`  raising ``KeyError`` when columns were :class:`MultiIndex` with only one level (:issue:`29749`)
 - Bug in :meth:`Series.__getitem__` and :meth:`DataFrame.__getitem__` raising blank ``KeyError`` without missing keys for :class:`IntervalIndex` (:issue:`27365`)
 - Bug in setting a new label on a :class:`DataFrame` or :class:`Series` with a :class:`CategoricalIndex` incorrectly raising ``TypeError`` when the new label is not among the index's categories (:issue:`38098`)
+- Bug in :meth:`Series.loc` and :meth:`Series.iloc` raising ``ValueError`` when inserting an array in a ``object`` Series of equal length (:issue:`37748`)
```

Comment:
array seems vague. Do you mean NumPy array, list, ExtensionArray (or selection of those)

Reply:
all of those i guess :)

Reply:
Indeed, np.array, list, tuple do cause the raise. From your example, ExtensionArray don't but also behave unexpectedly.